table/get_context.h

```diff
@@ -41,8 +44,15 @@ struct GetContextStats {
   uint64_t num_cache_compression_dict_bytes_insert = 0;
 };
 
+// A class to hold context about a point lookup, such as pointer to value
+// slice, key, merge context etc, as well as the current state of the
+// lookup. When a TableReader finds the internal key, it
+// calls SaveValue() to update the value for the key. This can happen
```

Comment:
Maybe move the part after "When a TableReader finds..." to somewhere closer to the implementation? Do you feel it might be a little too low-level for class comment?

Reply:
I'll try to reword it, but the intent is to describe how to use it, i.e the user must call SaveValue() for every matching key.

Reply:
In that case how about keep it where it is right now but reword it so that reader will know how to use it?